### PR TITLE
Test pipeline

### LINKED ISSUE #37: Add testing in the pipeline

### PR DESCRIPTION
**Pull request by JakubC-projects**
1. Adds automated testing to github build
    Also, I used caching to speed up building testing environment (shaves ca. 20s), not sure if it's worth the added complexity 
2. Adds 5 tests concerning token validation

closes #37

### CODE REVIEW

File: bcc-orgs/devops/test.docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    container_name: orgs_api_test
+    ports:
+      - "4000:4000"
+    volumes:
+      - ../src:/app/src
+      - ../db/migrations/:/db/migrations
+      - ../db/test-data/:/db/test-data
+      - ./docker-scripts/:/scripts
+    tty: true
+    env_file: ../.env.example
```
Comment (StevenMalaihollo) [inline bcc-orgs/devops/test.docker-compose.yml:18]: shouldnt we create a special .env.test file for testing purposes?

File: bcc-orgs/devops/test.docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    environment:
+      - AUTH0_ISSUER=${AUTH0_ISSUER}
+      - AUTH0_AUDIENCE=${AUTH0_AUDIENCE}
+      - AUTH0_CLIENT_ID=${AUTH0_CLIENT_ID}
+      - AUTH0_CLIENT_SECRET=${AUTH0_CLIENT_SECRET}
+    depends_on:
```
Comment (StevenMalaihollo) [inline bcc-orgs/devops/test.docker-compose.yml:24]: Shouldnt we place this service under orgs_db as it is dependant. Just reads a bit more logically i.m.o

File: bcc-orgs/src/auth/auth_test.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetJWKS(t *testing.T) {
+	GetKeySet(auth0Issuer)
```
Comment (StevenMalaihollo) [inline bcc-orgs/src/auth/auth_test.go:13]: This test is failing on my side: 

Failed to fetch key set: failed to fetch remote JWK: Get "https://https//bcc-sso-dev.eu.auth0.com/.well-known/jwks.json": dial tcp: lookup https on 127.0.0.11:53: read udp 127.0.0.1:53444->127.0.0.11:53: i/o timeout--- FAIL: TestGetJWKS (10.01s)

File: bcc-orgs/devops/test.docker-compose.yml (hunk 1 above)
Comment (JakubC-projects) [reply bcc-orgs/devops/test.docker-compose.yml:18]: Right makes sense

File: bcc-orgs/devops/test.docker-compose.yml (hunk 2 above)
Comment (JakubC-projects) [reply bcc-orgs/devops/test.docker-compose.yml:24]: Hmm, I think it's standard to put the most important stuff on the top, and push the rest to the bottom. Uncle Bob at least says, that it's better to put the more high-level function first and the the lower-level ones below them.

**Pull request merged**